src/main/groovy/com/bmuschko/gradle/docker/DockerExtension.groovy

```diff
@@ -66,6 +67,17 @@ class DockerExtension {
         dockerUrl
     }
 
+    public File getDefaultDockerCert() {
+        String dockerCertPath = System.getenv("DOCKER_CERT_PATH")
+        if(dockerCertPath) {
+            File certFile = new File(certPath);
+            if(certFile.exists() && this.url.startsWith('tcp://')) {
+                return certFile
+            }
+        }
+        return null;
+    }
+
```

Comment:
This looks better. Why do we only want to use this if the `url` scheme is `tcp`?

Reply:
> Why do we only want to use this if the url scheme is tcp?

it's a snippet from the original issue which was proposed as a workaround.
Also, I'm not sure, if we need a cert for unix socket or windows pipe.

Reply:
I'd remove that check as there is nothing stating it can't be used in those places. Other than that things LGTM.

Reply:
And for original comment that has since disappeared: I believe if we don't thrown an exception here, as part of this check, docker-java will eventually make a follow up check and thrown an exception there. My thinking was to fail as early as possible.

Reply:
>  docker-java will eventually make a follow up check and thrown an exception there

but since I return `null` if cert file does not exits, upstream will be happy and we'll proceed as if there were no cert at all, got docker barking, scratch our head and back to square #1 checking our cert.

It looks to me as premature optimization, which is not always good, you know. :)

Reply:
hahahaha yes yes I know ;) I'm not sure of the internals of `docker-java`, or how the check is handled, and if things are null will the check be skipped downstream? I seem to remember they will do an additional check, if no value/null is set, but its been a while since I looked at their codebase.